internal/objects/contour/validate.go

```diff
@@ -20,8 +20,8 @@ import (
 	"github.com/projectcontour/contour-operator/internal/slice"
 )
 
-// validateContour returns true if contour is valid.
-func validateContour(contour *operatorv1alpha1.Contour) error {
+// ValidateContour returns true if contour is valid.
+func ValidateContour(contour *operatorv1alpha1.Contour) error {
```

Comment:
I think there is an argument to be made that we put these helpers *in* the actual Contour object API specification. Yes, that means that we will have a GA behavior guarantee, but it also puts the *enforcement* of the API spec right next to the *definition* of the API spec.

Reply:
I don't see that the Contour operator will ever do anything other than configure Contour and Envoy installs, so we don't need to worry about exposing that we use a Contour controller and an Envoy install.